xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-default/src/main/java/org/xwiki/user/internal/UserReferenceConverter.java

```diff
@@ -42,13 +46,17 @@
     @Inject
     private UserReferenceResolver<String> userReferenceResolver;
 
+    @Inject
+    private Provider<XWikiContext> xcontextProvider;
+
     @Override
     protected UserReference convertToType(Type targetType, Object value)
     {
         if (value == null) {
             return null;
         }
 
-        return this.userReferenceResolver.resolve(value.toString());
+        WikiReference currentWikiReference = this.xcontextProvider.get().getWikiReference();
+        return this.userReferenceResolver.resolve(value.toString(), currentWikiReference);
```

Comment:
hmmm not sure this is correct. Shouldn't we use a `current` `UserReferenceResolver<String>` implementation instead?

Reply:
> Shouldn't we use a current UserReferenceResolver<String> implementation instead?

Actually this is already existing ;)

Reply:
Forget what I said, the fix is correct.

Reply:
Well there's only 1 potential improvement which is to get the current wiki from the ModelContext instead of from XWikiContext....

Reply:
Seems to be the right fix would be to provide a UserReferenceResolver<String> with hint "current" and use that like we do for DocumentReferenceResolver for example.

Reply:
Logic changed, but after some though I might have misinterpreted you explanations. Do you propose to introduce a new class implementing `UserReferenceResolver<String>`?

Reply:
Thomas said exactly the same thing I said at https://github.com/xwiki/xwiki-platform/pull/1386#discussion_r485602221. However, I checked the code and `UserReferenceResolver`  is working differently than entity reference resolvers.

Reply:
The hardest part would be to choose a name for the `current` hint class since `CurrentUserReferenceResolver`  already exists but it's a resolver for `CurrentUserReference` and not for `UserReference` ;)

Reply:
> Do you propose to introduce a new class implementing UserReferenceResolver<String>?

Yes.

Reply:
> The hardest part would be to choose a name for the current hint class since CurrentUserReferenceResolver already exists but it's a resolver for CurrentUserReference and not for UserReference ;) 

Actually we should call the existing class `CurrentUserReferenceUserReferenceResolver`. And the new one would be `CurrentUserReferenceResolver` (actually it would be `ConfiguredCurrentUserReferenceResolver` and forward to the impls).

Reply:
> Actually we should call the existing class CurrentUserReferenceUserReferenceResolver

And change the other impl for consistency ofc.

Reply:
> And the new one would be CurrentUserReferenceResolver

Actually it would be CurrentUserReferenceUserReferenceResolver... haha same name.. so it doesn't work for the naming part ;)

Reply:
I would name the new one (implementation of `UserReferenceResolver<String>` with hint "current") `CurrentStringUserReferenceResolver` (following CurrentStringDocumentReferenceResolver logic).